docs/migration/6_0.md

```diff
@@ -1,5 +1,25 @@
 # Technical Changes in Spartacus 6.0
 
+## SSR and Prerendering
+
+If you are using prerendering, you will need to import the following to your `AppServerModule`:
+
+```ts
+import { provideSsrAndPrerendering } from '@spartacus/setup/ssr';
+@NgModule({
+  ...
+  providers: [
+    ...provideSsrAndPrerendering({
+      serverRequestOrigin: process.env['SERVER_REQUEST_ORIGIN'],
+    }),
+    ...
+  ],
+})
+export class AppServerModule {}
+```
+
+The domain you specify `serverRequestOrigin` will be used as the origin when prerendering.
```

Comment:
Let's emphasize: when running live SSR (e.g. with ExpressJS), you don't want to configure `serverRequestOrigin`. It will be derived automatically from the client's request

Reply:
Ok, let's go with your proposal of implementation (to be more intuitive - let hardcoded value win a dynamic one). However IMHO hardcoded domain in SSR is not a common edge case, but rather unintended usage. Therefore I recommend to emphasize in .md docs and in JSDocs of this config property that it's recommended to use it in prerendering, but not in live SSR (as there the value will be figured out from the handled HTTP Request).

---
Side note:
IMHO we wouldn't have to discuss the implementation details of `provideServer()` for this edge case, if only we had a clear separation of concerns: SSR vs prerendering. E.g. have separate modules: AppServerModule and AppPrerenderingModule, which would contain different providers, only appropriate for their needs.

We now have an abstraction over them (SSR and prerendering), a shared `provideServer()` provider, and a shared AppServerModule. And hence we have to debate on unlikely edge-case handling and increasing the complexity of our internal code. 
On the other hand, having one module helps avoiding a complexity of setting up customer's repo 2 different modules: AppServerModule and AppPrerenderingModule (+ angular.json configurations).

We loose:
- simplicity of our implmentation, but not _that much_
- simplicity of our config (may suggest to configure the domain even for SSR, while it's confusing and makes little sense in practice, I believe)
We win:
- simplicty of customer's repo setup

When having to choose, I prefer simplicity for customers in this case.

Reply:
Thanks, I clarified in the docs.
And you're absolutely right, there are pros and cons to this approach, but I believe the balance is good in this case.